src/RF24LogAbstracts/Handler.cpp

```diff
@@ -0,0 +1,56 @@
+/**
+ * @file Handler.cpp
+ * @date Created 2 Oct 2020
+ * @author Witold Markowski (wmarkow)
+ * @copyright Copyright (C)
+ *      2020        Witold Markowski (wmarkow)
+ *      2021        Brendan Doherty (2bndy5) <br>
+ * This General Public License does not permit incorporating your program into
+ * proprietary programs.  If your program is a subroutine library, you may
+ * consider it more useful to permit linking proprietary applications with the
+ * library.  If this is what you want to do, use the GNU Lesser General
+ * Public License instead of this License.
+ */
+
+#include <RF24LogAbstracts/Handler.h>
+
+/* *************************** AbstractHandler defs **************************** */
+
+RF24LogAbstractHandler::RF24LogAbstractHandler()
```

Comment:
I'm not quite sure what are the C++ standards about class names vs header files names. This class is named ```RF24LogAbstractHandler``` but the header file is ```Handler.h```. Is it much better to understand where the class and header file have the same name?

Reply:
I was trying to avoid long winded filenames in even longer winded folders. I haven't checked with the C style guidelines yet, but I'm open to a more descriptive name.

Reply:
ok I renamed the "RF24Abstracts" folder as "RF24Parts" because not all the classes/files in their had strictly abstract structures. Also, I renamed _Stream.*_ and _Handler.*_ to _AbstractStream.*_ and _AbstractHandler.*_. Additionally I renamed some other files to make the casing be more uniform. 

Finally, I renamed _BaseHandler.h_ to _RF24BaseHandler.h_ and moved it back into the src folder. Log level descriptions exist in a LevelDescriptions.h file located in RF24Parts folder. As of this writing, all class names have not changed though. So far I've managed to get the builds all back to working order after this "refactor round 2" commit, but I'm getting a couple linker errors related to _RF24AbstractHandler.cpp_:
```
AbstractHandler.cpp.obj : error LNK2001: unresolved external symbol "public: virtual void __thiscall RF24LogBaseHandler::log(unsigned char,char const *,char const *,char * *)" (?log@RF24LogBaseHandler@@UAEXEPBD0PAPAD@Z)
AbstractHandler.cpp.obj : error LNK2001: unresolved external symbol "public: virtual void __thiscall RF24LogBaseHandler::setLogLevel(unsigned char)" (?setLogLevel@RF24LogBaseHandler@@UAEXE@Z)
```
I'm terrible with linker errors (its always a guessing game for me ☹️ ). The fact that this only shows up on Windows and not on Linux tells me its probably just another instance where I need to adjust the casing for a file being included somewhere (casing sensitivity in the Windows File System is damn weird -- sometimes it [accurately] ignores difference in casing and sometimes it doesn't)

Reply:
I have just built **GettingStarted/GettingStarted.ino** on Windows 10 against ESP32 and it compiles and links without any errors. Which example you want to build and against which platform? I could try it on my computer.

Reply:
I'm beyond Arduino over here. I'm talking about building the lib as a shared library and installing it on a PC (works fine on ubuntu but windows is getting linker errors).

Reply:
Ah, ok., I see. It would take me probably more time to set up the environment for such as case, so I will not help this time, sorry. But good to hear that you are using the library in this way. More developer scenarios (usages) will be tested and covered. Nice.

Reply:
I was able to remedy the linker errors (on windows builds) by making the the functions declared in RF24LogBaseHandler.h as pure virtual. This is fine because there are no definitions in the Base Handler (as the class is a base abstraction), and this change doesn't break the build for any platforms (re-tested Arduino and Ubuntu builds -- all checked out).

Next task is getting the examples to compile with the RF24Log lib that is installed prior to building the examples.